pkg/tfbridge/info.go

```diff
@@ -111,6 +114,31 @@ type ProviderInfo struct {
 	SkipExamples func(SkipExamplesArgs) bool
 }
 
+// DocRuleInfo controls file discovery and edits for any subset of docs in a provider.
+type DocRuleInfo struct {
+	// A list of edit rules to apply.
+	EditRules []DocsEdit
+
+	// A function to suggest alternative file names for a TF element.
+	AlternativeNames func(DocsPathInfo) []string
```

Comment:
It think I know what this does, but just because I've dug deep in the implementation before; for the casual users looking just at the docstrings this is all a bit laconic. Perhaps add a sentence or two about "file discovery". 

What may be happening is that we're trying to find markdown files for every resource and datasource inside a checkout of UpstreamRepoPath (or Repository). 

AlternativeNames here informs this process by making suggestions of possible file names to look for when finding docs for a given resource.

It's not obvious if those should have paths or not. Perhaps a quick example in the doc string?